connection/src/main/java/com/netcracker/Connection.java

```diff
@@ -0,0 +1,45 @@
+package com.netcracker;
+
+import com.netcracker.domain.Message;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.net.Socket;
+import java.net.SocketAddress;
+
+public class Connection implements Closeable {
+    private final Socket socket;
+    private final ObjectOutputStream outputStream;
+    private final ObjectInputStream inputStream;
+
+    public Connection(Socket socket) throws IOException {
+        this.socket = socket;
+        this.outputStream = new ObjectOutputStream(socket.getOutputStream());
+        this.inputStream = new ObjectInputStream(socket.getInputStream());
+    }
+
+    public void send(Message message) throws IOException {
+        synchronized (outputStream) {
```

Comment:
&#x1F34E; **Issue**
Do you mean to use a connection object from different threads ? What for ? I think the code would be much simpler if you limit the number of places where you synchronize threads in order to do it efficiently and avoid deadlocks.